src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/AbstractProjectConfigurationDefineConstantsEncoding.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug
+{
+    /// <summary>
+    ///  Converts the single string representation to multiple key, value pairs used
+    ///  for defining constants.
+    /// </summary>
+    internal static class AbstractProjectConfigurationDefineConstantsEncoding
```

Comment:
This class can be non-static, as everything inside can be instance fields and just instantialize in the value provider constructor